blog/02-02-2023-connect-and-interact-with-celo-using-web3onboard-library/connect-and-interact-with-celo-using-web3onboard-library.md

```diff
@@ -0,0 +1,1106 @@
+            {functionName === "stakeOnBehalf" && <TextField margin="normal" required fullWidth id="text" label="Account to stake for" name="amount" autoComplete="account" type={"text"} autoFocus sx={{ border: `0.1em solid ${blue[900]}`, borderRadius: "5px" }} style={{ color: "whitesmoke" }} onChange={(e) => handleAccountChange(e)} />}
+            <Button
+              type="submit"
+              fullWidth
+              variant="contained"
+              sx={{
+                mt: 3,
+                mb: 2,
+                height: '100px',
+                fontWeight: "bold",
+                display: 'flex',
+                justifyContent: 'center',
+                alignItems: 'center',
+                background: purple[300]
+              }}
+            >
+              { loading? <span>Trxn in Progress ... <Spinner color={"white"} /></span> : functionName }
+            </Button>
+          </Box>
+        </Box>
+      </Container>
+      <Footer sx={{ mt: 8, mb: 4 }} />
+    </React.Fragment>
+  );
+}
+
+```
+
+## components/api/contractdata.ts
+
+Update this file with the latest contract addresses from the last deployment we made.
+
+```bash
+import vault from "../../../foundry/out/Vault.sol/Vault.json";
+import token from "../../../foundry/out/RewardToken.sol/RewardToken.json";
+
+export default function getContractData() {
+  return {
+    vaultAbi: vault.abi,
+    tokenAbi: token.abi,
+    vaultAddr: "0x89330624480dE7Fc2e651233B022B85a1F998de7", //<===
+    tokenAddr: "0xefA906f63ea950318d8d5Af13ae2E5D2aC221Fe4" //<===
+  }
+}
+
+```
+
+## components/api/index.ts - (changes)
+
+- Import required modules/files. Replace all imports in the current 'api/index.ts' file with ones below.
+
+> Note: Web3Onboard returns Ethereum's EIP1193 compatible provider which we will wrap in etherjs's to get the web3 provider we need to broadcast transaction to Celo's testnet.
+
+```bash
+import getContractData from "./contractdata"; 
+import { InstanceProps, OptionProps, Profile, transactionResult } from "@/interfaces";
+import { ethers, Contract, ContractReceipt } from "ethers";
+import { EIP1193Provider } from "@web3-onboard/core";
+```
+
+- Create a function that accepts any valid external provider and return a valid web3 provider.
+
+```bash
+// Return ether's web3 provider
+function wrappedProvider(provider: EIP1193Provider) {
+  if (!provider) alert('Provider not ready');
+  return new ethers.providers.Web3Provider(provider, 'any');
+}
+```
+
+- The next function returns contract instances with provider and signer capabilities. A contract instance with provider property present can only allow us read from the blockchain but not to send transaction capable of modifying the state of the blockchain while an instance with signer capability can read from the blockchain and modify it through valid transaction.
+
+> Note: Contract instance with provider property is enough to read from the blockchain.
+
+```bash
```

Comment:
Isn't this has to be ```js?

Reply:
Please read the guidelines for review. This is the correct format.